### PR TITLE
UK SSO: adapt login/signup to common flow provided in external_auth.

### PR DESCRIPTION
**Pull request by sergiitk**
#### What's this PR do?
- Adapts Dosomething UK module: use login/signup provided in [`external_auth`](/dosomething/drupal-external-auth)
#### How should this be manually tested?
- Enable and setup Dosomething UK module
- Run `ds test uk`  
  image
#### What are the relevant tickets?
- Part of [#DS-379](https://jira.dosomething.org/browse/DS-379): Move functionality common to UK and Canada to the new nodule.
- dosomething/drupal-external-auth#2

### CODE REVIEW

File: lib/modules/dosomething/dosomething_uk/dosomething_uk.module [hunk 1]
```diff
@@ -35,326 +35,23 @@ function dosomething_uk_preprocess_page(&$vars) {
...
- * @param string $email
- *   The email of the user.
- * @param string $password
- *   The password of the user.
- * @param DosomethingUkSsoController $sso
- *   The SSO controller.
- *
- * @return DosomethingUkSsoUser
- *   The remote user object.
+ * Implements external_auth_method().
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_uk/dosomething_uk.module]: Picky, but should read as `Implements hook_external_auth_method()` right?

File: lib/profiles/dosomething/drupal-org.make [hunk 2]
```diff
@@ -200,6 +200,12 @@ projects[xmlsitemap][subdir] = "contrib"
 
 ; GIT PROJECTS
 
+; External Auth
+projects[external_auth][type] = "module"
+projects[external_auth][download][type] = "git"
+projects[external_auth][download][url] = "https://github.com/DoSomething/drupal-external-auth.git"
+projects[external_auth][subdir] = "contrib"
```
Comment (aaronschachter) [inline lib/profiles/dosomething/drupal-org.make:208]: Up to you, but you may want to consider adding the specific commit to pull down from here.  Fine to leave as is too, to not worry about it... but, just need to be careful about breaking UK.

**Comment by aaronschachter** [comment]
Minor comments but :+1:  to ship

File: lib/modules/dosomething/dosomething_uk/dosomething_uk.module (hunk 1 above)
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_uk/dosomething_uk.module]: Sure, my mistake. Will fix it.

File: lib/profiles/dosomething/drupal-org.make (hunk 2 above)
Comment (sergiitk) [inline lib/profiles/dosomething/drupal-org.make:208]: That's temporary solution, I'm planning to release it to drupal.org when it's ready.
I'll fix the commit though, just to make it more explicit.

**Pull request merged**